packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -1812,6 +1812,149 @@ import 'output-localization-file_en.dart' deferred as output-localization-file_e
       });
     });
 
+    group('select messages', () {
+      testWithoutContext('should throw attempting to generate a select message without placeholders', () {
+        const String selectMessageWithoutPlaceholdersAttribute = '''
+{
+  "genderSelect": "{gender, select, female {She} male {He} other {they} }",
+  "@genderSelect": {
+    "description": "Improperly formatted since it has no placeholder attribute."
+  }
+}''';
+
+        final Directory l10nDirectory = fs.currentDirectory.childDirectory('lib').childDirectory('l10n')
+          ..createSync(recursive: true);
+        l10nDirectory.childFile(defaultTemplateArbFileName)
+            .writeAsStringSync(selectMessageWithoutPlaceholdersAttribute);
+
+        expect(
```

Comment:
This is a pretty complicated expect statement. It's fine but I took a crack at the indentation anyway. If you think there's a marginal improvement then here and below.
```dart
expect(
  () {
    LocalizationsGenerator(
      fileSystem: fs,
      inputPathString: defaultL10nPathString,
      outputPathString: defaultL10nPathString,
      templateArbFileName: defaultTemplateArbFileName,
      outputFileString: defaultOutputFileString,
      classNameString: defaultClassNameString,
    )
      ..loadResources()
      ..writeOutputFiles(BufferLogger.test());
  },
  throwsA(
    isA<L10nException>().having(
      (L10nException e) => e.message,
      'message',
      contains('Check to see if the select message is in the proper ICU syntax format'),
    ),
  ),
);
```

Reply:
Honestly, I wouldn't have written it myself with my current knowledge. This is a modified version of the already existing tests.

Indentation was changed.